Rubberduck.CodeAnalysis/Inspections/Concrete/BooleanAssignedInIfElseInspection.cs

```diff
@@ -42,33 +39,28 @@ namespace Rubberduck.Inspections.Concrete
     /// </example>
     public sealed class BooleanAssignedInIfElseInspection : ParseTreeInspectionBase
     {
-        public BooleanAssignedInIfElseInspection(RubberduckParserState state)
-            : base(state) { }
+        public BooleanAssignedInIfElseInspection(IDeclarationFinderProvider declarationFinderProvider)
+            : base(declarationFinderProvider)
+        {}
         
         public override IInspectionListener Listener { get; } =
             new BooleanAssignedInIfElseListener();
 
-        protected override IEnumerable<IInspectionResult> DoGetInspectionResults()
+        protected override string ResultDescription(QualifiedContext<ParserRuleContext> context)
         {
-            return Listener.Contexts
-                .Select(result => new QualifiedContextInspectionResult(this,
-                                                       string.Format(InspectionResults.BooleanAssignedInIfElseInspection,
-                                                            (((VBAParser.IfStmtContext)result.Context).block().GetDescendent<VBAParser.LetStmtContext>()).lExpression().GetText().Trim()),
-                                                       result));
+            var literalText = ((VBAParser.IfStmtContext) context.Context)
```

Comment:
That's not a crazy idea at all. Actually there's something odd about losing all benefits of generics here... it's almost like.. like it wants to be `string ResultDescription<TContext>(QualifiedContext<TContext> context) where TContext : ParserRuleContext`?

Reply:
I already have an idea how to wire up a generic version. However, this will need rework on all parse tree inspections and on the interface and its users as well.

Reply:
Since the latest push, the parse tree inspection base classes and the listeners are generic with respect to the context type.

Reply:
I think the latest changes is a vast improvement, nicely done! The only nitpick I have is that a select number of inspections use the general `ParserRuleContext` as the `TContext` due to the fact that they work with more than one contexts. Ideally, it would be possible to statically enumerate valid contexts they can have but I don't see a solution for that short of adding marker interfaces to various parser rule contexts. For this PR, I'm happy with the direction you have taken.